xarray/tests/duckarrays/base/strategies.py

```diff
@@ -0,0 +1,161 @@
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+
+import xarray as xr
+from xarray.core.utils import is_dict_like
+
+from . import utils
+
+all_dtypes = (
+    npst.integer_dtypes()
+    | npst.unsigned_integer_dtypes()
+    | npst.floating_dtypes()
+    | npst.complex_number_dtypes()
+)
+
+
+def numpy_array(shape, dtypes=None):
+    if dtypes is None:
+        dtypes = all_dtypes
+
+    def elements(dtype):
+        max_value = 100
+        min_value = 0 if dtype.kind == "u" else -max_value
+
+        return npst.from_dtype(
+            dtype, allow_infinity=False, min_value=min_value, max_value=max_value
+        )
+
+    return dtypes.flatmap(
+        lambda dtype: npst.arrays(dtype=dtype, shape=shape, elements=elements(dtype))
+    )
+
+
+def dimension_sizes(min_dims, max_dims, min_size, max_size):
+    sizes = st.lists(
+        elements=st.tuples(st.text(min_size=1), st.integers(min_size, max_size)),
```

Comment:
My understanding of this is that hypothesis will also try creating xarray dims with all sorts of different names.

Is that actually necessary? What duck array behaviour could possibly break dependent on the name of the dimensions of the xarray object wrapping the duck array? 

Couldn't we speed up the hypothesis tests by always using dimensions named `"dim_0"`, `"dim_1"` etc. instead?

Reply:
nothing, probably, I think I was trying to write general `xarray` strategies. Creating fixed names might be a bit faster, but since we don't reject any examples using `.filter` or `assume` I don't think it would actually be that much faster.

Reply:
> I was trying to write general xarray strategies.

If we're exposing these publicly then varying dimension names makes sense, so I retract this comment!